### PR TITLE
fence_raritan_px3: new fence agent

### PR DESCRIPTION
**Pull request by watologo1**
Tested on model PX3-5466V with firmware version: 3.6.0.5

### CODE REVIEW

**Comment by watologo1** [comment]
While I tested these features to work:
- on/off/status/list/list-status
- single outlets/outletgroups

I run into following issues:
- I had 2 systems, one had rather long network latencies and it costed me quite some time, that it's a simple timeout I was running into while it worked the some time before. The other system was much faster, still I increased the default timeout as it's really nasty to struggle with such an issue.
- Doing on/off switching, one gets asked whether you really want to do it. I searched the config, but could not find anything to switch that off. Sending 'y' works and also the next prompt is correctly "expected".  Means I see the XXXXX debug message I left in there immediately:
```
        print("YYYYY")
        conn.log_expect(options["--command-prompt"], int(options["--shell-timeout"]))
        print("XXXXXXXX")
```
Then I it always hangs until the connection times out.
I tried to additionally expect a ".*" to get through without success.
Some help/hint would be appreciated what/why it hangs there.

Thanks.

File: agents/raritan_px3/fence_raritan_px3.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+		# fetch state of previously parsed outlet from next line/iter
+		if len(split_line) == 3 and split_line[0] == "Power" and split_line[1] == "state:":
+			outlet_state = split_line[2]
+			FenceRaritanPX3.outlets[outlet_no] = [outlet_name, outlet_state]
+	logging.debug("Outlets found:\n%s", FenceRaritanPX3.outlets)
+	return FenceRaritanPX3.outlets
+
+def get_power_status(conn, options):
+
+	if FenceRaritanPX3.plug:
```
Comment (oalbrigt) [inline agents/raritan_px3/fence_raritan_px3.py:107]: What status's does it report? on/off/unknown is what the fencing library expects.

Comment (watologo1) [reply agents/raritan_px3/fence_raritan_px3.py:107]: Yes, on/off

Originally On/Off as shown in the comment:
                Power state: On
(credentials removed):

I save all in an outlet dict, if necessary a second call also
storing outletgroup data is done:

 ./fence_raritan_px3 ... --plug=20  --action=list-status
1,,On
2,,On
3,,On
4,,On
5,,On
6,,On
7,,On
8,Testung,On
9,,On
10,,On
11,,On
12,,On
13,,On
14,mustang1,On
15,mustang2,On
16,mustang4,On
17,mustang3,On
18,,On
19,,On
20,,On

Some plugs/outlets do have names, some can only be accessed via ID.
Raw dict:
{'1': ['', 'On'], '2': ['', 'On'], '3': ['', 'On'], '4': ['', 'On'], '5': ['', 'On'], '6': ['', 'On'], '7': ['', 'On'], '8': ['Testung', 'On'],
'9': ['', 'On'], '10': ['', 'On'], '11': ['', 'On'], '12': ['', 'On'], '13': ['', 'On'], '14': ['mustang1', 'On'],
'15': ['mustang2', 'On'], '16': ['mustang4', 'On'], '17': ['mustang3', 'On'], '18': ['', 'On'], '19': ['', 'On'], '20': ['', 'On']}

I could (should?) already convert the uppercase to lower?
I do this later:
        if FenceRaritanPX3.plug:
                return FenceRaritanPX3.outlets[str(FenceRaritanPX3.plug)][1].lower()
        elif FenceRaritanPX3.group_name:
                return FenceRaritanPX3.outletgroups[FenceRaritanPX3.group_name][1].lower()

So on/off can be returned.

Comment (oalbrigt) [reply agents/raritan_px3/fence_raritan_px3.py:107]: Sounds good.

Yeah. That makes sense, to make it the same as the other agents, which usually maps the statuses.

Example:
https://github.com/ClusterLabs/fence-agents/blob/master/agents/vmware_rest/fence_vmware_rest.py#L14

**Pull request merged**